packages/client/src/app/jobs/executeJobModal.tsx

```diff
@@ -1,17 +1,24 @@
+import React, { useState, useMemo, useCallback } from 'react';
 import { Button, Modal, Table, Text } from '@nextui-org/react';
 import { MitmControlDTO } from '@rotom/connections';
 import { Selection } from '@react-types/shared/src/selection';
 import { toast } from 'react-toastify';
-import { useCallback, useState } from 'react';
+
+import { SearchInput } from '../status/search';
 
 interface ExecuteJobModalProps {
   closeModal: () => void;
   devices?: MitmControlDTO[];
   jobId: string;
 }
 
-export const ExecuteJobModal = ({ closeModal, devices, jobId }: ExecuteJobModalProps): JSX.Element => {
+export const ExecuteJobModal: React.FC<ExecuteJobModalProps> = ({ closeModal, devices, jobId }) => {
   const [selectedDevices, setSelectedDevices] = useState<Selection>();
+  const [search, setSearch] = useState('');
+
+  const filteredDevices = useMemo(() => {
+    return devices?.filter((device) => device.deviceId?.includes(search) || device.origin?.includes(search)) || [];
+  }, [devices, search]);
```

Comment:
Just tested it, and there is a small issue: it's sensible to the case. It could be cool if it wasn't. One of the way to do it:
```suggestion
  const filteredDevices = useMemo(() => {
    return (
      devices?.filter(
        (device) =>
          device.deviceId?.toLowerCase().includes(search.toLowerCase()) ||
          device.origin?.toLowerCase().includes(search.toLowerCase()),
      ) || []
    );
  }, [devices, search]);
```
You could also prevent transforming multiple times to lower case the same variable by saving the lower case value to an intermediary value.